tests/vector.rs

```diff
@@ -33,6 +33,109 @@ fn test_from_value() {
     assert_eq!(Vector4::from_value(76.5f64), Vector4::new(76.5f64, 76.5f64, 76.5f64, 76.5f64));
 }
 
+macro_rules! test_op_add {
+    ($VectorN:ident { $($field:ident),+ }, $s:expr, $v:expr) => (
+        // vector + vector ops
+        assert_eq!($v + $v, $VectorN::new($($v.$field + $v.$field),+));
```

Comment:
> I wonder if this is just re-testing the implementation? Dunno if it matters though...

Could you clarify, do you mean versus an alternative of comparing with a know good result, or something else? If it's any help, I did catch some copy and paste errors I made in some of the operator implementations with this test :)